presto-main/src/main/java/com/facebook/presto/sql/SqlPath.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+
+public class SqlPath
+{
+    private List<SqlPathElement> parsedPath;
+    private String rawPath;
+
+    @JsonCreator
+    public SqlPath(@JsonProperty("rawPath") String path)
+    {
+        this.rawPath = path;
+    }
+
+    @JsonProperty
+    public String getRawPath()
+    {
+        return rawPath;
+    }
+
+    public List<SqlPathElement> getParsedPath()
+    {
+        if (parsedPath == null) {
+            parsePath();
+        }
+
+        return parsedPath;
+    }
+
+    private void parsePath()
+    {
+        //TODO: add further verification once default path is determined
+        ImmutableList.Builder<SqlPathElement> builder = ImmutableList.builder();
+        //avoid splitting on commas contained within quotes as part of catalog/schema names
+        String[] schemas = rawPath.split(",(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)");
+
+        for (String schema : schemas) {
+            String catalogName;
+            String schemaName;
+            SqlPathElement element;
+
+            //avoid splitting on periods contained within quotes as part of catalog/schema names
+            String[] individual = schema.split("\\.(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)");
+
+            if (individual.length > 1) {
+                catalogName = individual[0];
+                schemaName = individual[1];
+                element = new SqlPathElement(catalogName, schemaName);
+            }
+            else {
+                element = new SqlPathElement(schema);
```

Comment:
We can use `new SqlPathElement(null, schema);` and we can get rid of this constructor.